config/cloud-config/node-config-template.yaml

```diff
@@ -73,19 +73,28 @@ write_files:
     permissions: 0755
     content: |
       #! /usr/bin/bash
-      
+      CALICO_VERSION=$(/usr/bin/etcdctl get /calico-install/version) >/dev/null 2>&1 || CALICO_VERSION="0.12.0"
```

Comment:
Ah, so the idea is that each node will use its own local `etcd` instance to determine which version to run (so you can mix and match Calico versions) rather than each node looking to the master `etcd` to determine which Calico version to use?